rfd/0xxx-kube-mfa-sessions.md

```diff
@@ -0,0 +1,142 @@
+
+Local proxy will generate ephemeral kubeconfig that kubectl will use and then based on SNI proxy will select real user credentials
+to access remote Kubernetes service. Existing user's kubeconfig (including entries from `tsh kube login`) will be integrated into
+this ephemeral kubeconfig for better UX.
+
+Chart from RFD_NNN demonstrating local Kubernetes proxy flow:
+```
+ ┌───────┐                                        ┌─────────┐                         ┌───────────────┐
+ │kubectl│                                        │tsh local│                         │Teleport Proxy/│
+ │       │                                        │  proxy  │                         │ Load Balancer │
+ └┬──────┘                                        └─┬───────┘                         └────────────┬──┘
+  │                                                 │                                              │
+  │                                                 ├──┐                                           │
+  │                                                 │  │generate:                                  │
+  │                                                 │  │local CA                                   │
+  │                                                 │  │local credentials                          │
+  │                                                 │◄─┘ephemeral KUBECONFIG                       │
+  │                                                 │                                              │
+  │ server https://localhost:8888                   │                                              │
+  │ sni my-kube.kube-teleport-local.my-teleport.com │                                              │
+  │ local credentials                               │                                              │
+  ├────────────────────────────────────────────────►│ server https://my-teleport.com:443           │
+  │                                                 │ sni kube-teleport-proxy-alpn.my-teleport.com │
+  │                                                 │ "real" user credentials                      │
+  │                                                 ├─────────────────────────────────────────────►│
+  │                                                 │                                              │
+  │                                                 │◄─────────────────────────────────────────────┤
+  │◄────────────────────────────────────────────────┤                                              │
+  │                                                 │                                              │
+```
+
+User experience will look like this:
+
+```bash
+$ tsh proxy kube test-kube-cluster -p 8888
+Preparing the following Teleport Kubernetes clusters:
+Teleport Cluster Name Kube Cluster Name
+--------------------- ----------------------
+my-teleport.com       test-kube-cluster
+
+Enter an OTP code from a device:
```

Comment:
The current implementation of kube local proxy supports multiple kube clusters. It's convenient if the user wants to log in to multiple clusters at the same time but I believe the MFA request is per kube cluster as well.  Are we prompting multiple times in that case?